controller/bindpoints/overlay.go

```diff
@@ -0,0 +1,153 @@
+/*
+	Copyright NetFoundry Inc.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+	https://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+package bindpoints
+
+import (
+	"crypto/tls"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/michaelquigley/pfxlog"
+	"github.com/openziti/identity"
+	"github.com/openziti/sdk-golang/ziti"
+)
+
+// IdentityConfig represents the BindPointConfig when an identity is supplied as opposed to an address
+type OverlayBindPoint struct {
```

Comment:
Add an interface enforce:

`var _ xweb.BindPoint = (*OverlayBindPoint )(nil)


It will ensure that if the interface changes, it visibly breaks in IDEs.